extensions/s3/src/test/java/io/deephaven/extensions/s3/SingletonContainers.java

```diff
@@ -0,0 +1,87 @@
+/**
+ * Copyright (c) 2016-2023 Deephaven Data Labs and Patent Pending
+ */
+package io.deephaven.extensions.s3;
+
+import org.testcontainers.containers.MinIOContainer;
+import org.testcontainers.containers.localstack.LocalStackContainer;
+import org.testcontainers.containers.localstack.LocalStackContainer.Service;
+import org.testcontainers.utility.DockerImageName;
+import software.amazon.awssdk.auth.credentials.AwsBasicCredentials;
+import software.amazon.awssdk.auth.credentials.StaticCredentialsProvider;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.s3.S3Client;
+
+import java.net.URI;
+
+final class SingletonContainers {
+
+    // This pattern allows the respective images to be spun up as a container once per-JVM as opposed to once per-class
+    // or once per-test.
+    // https://java.testcontainers.org/test_framework_integration/manual_lifecycle_control/#singleton-containers
+    // https://testcontainers.com/guides/testcontainers-container-lifecycle/#_using_singleton_containers
+
+    static final class LocalStack {
+        private static final LocalStackContainer LOCALSTACK_S3 =
+                new LocalStackContainer(DockerImageName.parse("localstack/localstack:3.1.0")).withServices(Service.S3);
```

Comment:
I'm happy to discuss wider plans for bumping versions and getting bot PRs for such actions. Unfortunately, even if this was decoratively handled in a docker-compose.yml file, GH dependabot doesn't natively support docker-compose.yml yet https://github.com/dependabot/dependabot-core/issues/390. It may be worthwhile taking a look at https://github.com/renovatebot/renovate which seems to do a lot of this type of work.

Reply:
Let's create a follow-up issue, to address this as well as "hidden in groovy" dependencies, etc.

Reply:
https://github.com/deephaven/deephaven-core/issues/5115